static/js/narrow.js

```diff
@@ -525,6 +525,23 @@ function pick_empty_narrow_banner() {
     var first_operator = first_term.operator;
     var first_operand = first_term.operand;
 
+    var stream_name = narrow.stream();
+    var sub = stream_data.get_sub(stream_name);
+
+    // Checks if narrowed to never subscribed stream
+    var never_sub = false;
+    var i;
+
+    _.each(page_params.neversubbed_info, function (stream) {
```

Comment:
I think this would be a lot cleaner if written as a functional style:
```
var never_sub = _.some(page_params.neversubbed_info, function (stream) {
    return stream.name == stream_name;
});